### ISSUE TITLE
Support line ranges

### ISSUE THREAD

**Issue by mre** [mre/cargo-inspect#10]
```
cargo inspect --range 5,9
```
should work and print the expanded code for the lines 5 to 9.

**Comment by jonathansty** [comment]
I had a quick look at this as this seemed like an easy issue to start with. 
Although after adding the command line parameter to specify a range I ran into the issue that the output that's received from the rustc compiler potentially has more lines added meaning the line numbers won't match with the input data. 

The easiest example of a case for this is the insertion of the default std stuff: 
```
#[prelude_import]
use std::prelude::v1::*;
#[macro_use]
extern crate std;
```
but there's other cases where new lines are inserted as well.  

Basically I would need a way to map the input line numbers to output regions to make sure that everything that got expanded in the original line range is being outputted correctly.

**Comment by mre** [comment]
Yeah I think it's surprisingly hard to do. I thought about adding a special comment with the line number above the expanded code to make that easier (similar to the `verbose` option). Alternatively we could add support for filters, similar to cargo expand: https://github.com/dtolnay/cargo-expand/blob/master/src/filter.rs
